addons/website/tools.py

```diff
@@ -71,6 +71,8 @@ def MockRequest(
         request.httprequest.url = werkzeug.urls.url_join(url_root, path)
     if website:
         request.website_routing = website.id
+    if country_code and odoo.http.geoip2:
+        request.geoip._city_record = odoo.http.geoip2.models.City({'country': {'iso_code': country_code}})
```

Comment:
+ remove the now useless line in the DotDict for the session
```suggestion
    if country_code:
        request.geoip._city_record = odoo.http.geoip2.models.City({'country': {'iso_code': country_code}})
```